Update README.md

slrslr opened the pull request:
Fix non working command and extend it.

slrslr commented:
Yes, it started looking too long. [Here is yours requested bash script Zeronet-Conservancy-Linux-installer-updater.sh](https://pastebin.com/JzRFkSqu). Please place it where it should be located. I think that the instructions may be something like:


#### General Linux installer and updater bash script
You can run following command to install or update ZeroNet-Conservancy on Linux:
`curl URLHERE|bash`

caryoscelus commented:
@slrslr if you want this to be "official" script, I have a few issues we'll have to resolve first:

- data directory is considered inner detail, like with any "similar" apps; there's no need to expose it to user except if they want to, in which case they can ln themselves or just user different --start-dir
- there's more packages listed in README as requirements for apt-based distros; have you checked that they are no longer required (they are omitted in script)?
- have you tested various distros mentioned in script?
- note that --no-migrate will be default as soon as I test it, so probably there's not gonna be need in it by the time we resolve other issues

if you'd rather keep this script as your own, feel free to add it to your repo and I'll add a link to it

> definitely above Nix instructions because Nix is used by minority of users

which users? nix is the most supported system since I'm developing on it and there's a maintainer for nix package; it's possible to use nix as package manager on any sane system (maybe even windows+wsl these days). if there would be maintainers for packages on other platforms, I'd consider moving them further on top, but as is, it doesn't feel right to me; generally I'll make sure docker works and make appimage around v0.8 maturity, those probably will be on top

slrslr commented:
Second version of the script: https://pastebin.com/F9sMiTNf

.
.

> @slrslr if you want this to be "official" script, I have a few issues we'll have to resolve first:
> 
>     * data directory is considered inner detail, like with any "similar" apps; there's no need to expose it to user except if they want to, in which case they can ln themselves or just user different --start-dir

If I understand correctl,y you do not want current default data directory to be used like this in the script:
ln -s "$HOME/.local/share/zeronet-conservancy/"{znc.conf,data,log} "$zndir"/ 2>/dev/null

so I have removed it from my second version of the script (linked above), but it will make it harder for some people to find out Zeronet data/configuration.

> 
>     * there's more packages listed in README as requirements for apt-based distros; have you checked that they are no longer required (they are omitted in script)?

ZNC worked for me with mine used packages, but I have inserted all APT packages [mentioned in README](https://github.com/zeronet-conservancy/zeronet-conservancy#apt-based-debian-ubuntu-etc) into second version of the bash script.

> 
>     * have you tested various distros mentioned in script?

No, IIRC the mine used commands comes from some good rated reply on a Stack exchange on how to install Python3 venv, pip. If anything does not work, Issue can be reported and packages added/removed from the bash script.

> 
>     * note that --no-migrate will be default as soon as I test it, so probably there's not gonna be need in it by the time we resolve other issues
> 

I have removed --no-migrate from the second version of the script.

> 
> if you'd rather keep this script as your own, feel free to add it to your repo and I'll add a link to it
>

I would prefer it is managed in ZNC repo, that way you and other repo users have also more control on the script content than linking it.
 
>
> > definitely above Nix instructions because Nix is used by minority of users
> 
> which users? nix is the most supported system since I'm developing on it and there's a maintainer for nix package; it's possible to use nix as package manager on any sane system (maybe even windows+wsl these days). if there would be maintainers for packages on other platforms, I'd consider moving them further on top, but as is, it doesn't feel right to me; generally I'll make sure docker works and make appimage around v0.8 maturity, those probably will be on top

My point was that the instructions for the most widely used platforms should be on the top of the list or there be a clickable index of platforms so person can click instead of scrolling through already long list of "How to join".
